src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -674,13 +674,12 @@ static Value builtinArraySlice(ExecutionState& state, Value thisValue, size_t ar
     int64_t n = 0;
     // Let count be max(final - k, 0).
     // Let A be ArraySpeciesCreate(O, count).
-    Value val = arraySpeciesCreate(state, thisObject, std::max(((int64_t)finalEnd - (int64_t)k), (int64_t)0));
-    ArrayObject* array = val.asObject()->asArrayObject();
```

Comment:
How about just simply removing the invalid `asArrayObject()` cast?

Reply:
Valid point. Fixed it